lib/modules/versioning/nuget/version.ts

```diff
@@ -0,0 +1,94 @@
+import { regEx } from '../../../util/regex';
+import type { NugetVersion } from './types';
+
+// We can't just use `Number.parseInt()` beacause it parses `123abc` as `123`,
+// which leads to incorrect version comparison.
+function ensureNumber(input: string): number | null {
+  if (!regEx(/^\d+$/).test(input)) {
+    return null;
+  }
+
+  return Number.parseInt(input, 10);
+}
+
+function comparePrereleases(x: string, y: string): number {
+  const xParts = x.split('.');
+  const yParts = y.split('.');
+
+  const maxLen = Math.max(xParts.length, yParts.length);
+  for (let i = 0; i < maxLen; i += 1) {
+    const xPart = xParts[i] ?? '';
+    const yPart = yParts[i] ?? '';
+    const xNum = ensureNumber(xPart);
+    const yNum = ensureNumber(yPart);
+
+    const res =
+      xNum !== null && yNum !== null
+        ? Math.sign(xNum - yNum)
+        : xPart.localeCompare(yPart, undefined, { sensitivity: 'base' });
```

Comment:
wonder how it knows to correctly sort `1.1-b9` and `1.1-b10` 🤔

Reply:
[`1.1-b9` > `1.1-b10`](https://nugettools.azurewebsites.net/6.11.0-preview.2/version-comparison?versionA=1.1-b10&versionB=1.1-b9)

Reply:
🤦‍♂️

Reply:
Yes, we have to split prerelease part by dot, and there is no implicit letter-number/number-letter split. That's why here we have custom comparator at all.

Reply:
I now remember why I used `beta001` in our projects until all tooling was semver v2 compatible.